docs/sdk-and-tools/sdk-py/sdk-py-cookbook.md

```diff
@@ -236,34 +227,53 @@ transaction = transfer_factory.create_transaction_for_esdt_token_transfer(
     token_transfers=[first_transfer, second_transfer]
 )
 
-print("Transaction:", transaction.__dict__)
-print("Transaction data:", transaction.data)
+print("Transaction:", transaction_converter.transaction_to_dictionary(transaction))
+print("Transaction data:", transaction.data.decode())
 ```
 
-[comment]: # (mx-context-auto)
+### Decoding Transactions
+
+For example, when sending multiple ESDT and NFT tokens, the receiver field of the transaction is the same as the sender field and also the value is set to `0` because all the information is encoded in the `data` field of the transaction.
+
+For decoding the data field we have a so called `TransactionDecoder`. We fetch the transaction from the network and then use the decoder.
+
+```py
+from multiversx_sdk.network_providers import (ProxyNetworkProvider,
+                                              TransactionDecoder)
+
+proxy = ProxyNetworkProvider("https://devnet-api.multiversx.com")
+transaction = proxy.get_transaction("3e7b39f33f37716186b6ffa8761d066f2139bff65a1075864f612ca05c05c05d")
+
+decoder = TransactionDecoder()
+decoded_transaction = decoder.get_transaction_metadata(transaction)
+
+print(decoded_transaction.to_dict())
+```
 
 ## Relayed Transactions
 
 First, we get the newtwork configuration using the network providers.
 
-```
-from multiversx_sdk_network_providers import ProxyNetworkProvider
+```py
+from multiversx_sdk.network_providers import ProxyNetworkProvider
 
 provider = ProxyNetworkProvider("https://devnet-gateway.multiversx.com")
 network_config = provider.get_network_config()
 ```
 
-[comment]: # (mx-context-auto)
-
 ### Relayed V1
 
-```
+```py
 from pathlib import Path
-from multiversx_sdk_core.transaction_builders.relayed_v1_builder import RelayedTransactionV1Builder
-from multiversx_sdk_core import Transaction, TransactionComputer, Address
-from multiversx_sdk_wallet.user_signer import UserSigner
 
-signer = UserSigner.from_pem_file(Path("./testwallets/bob.pem"))
+from multiversx_sdk.core import Address, Transaction, TransactionComputer
+from multiversx_sdk.core.transactions_factories.relayed_transactions_factory import \
```

Comment:
Can we have simpler imports here?

Reply:
Regenerated the cookbook. Imports look like this:
```py
from multiversx_sdk import ...
```